librz/include/rz_cmp.h

```diff
@@ -0,0 +1,29 @@
+// SPDX-FileCopyrightText: 2021 Dhruv Maroo <dhruvmaru007@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#ifndef RZ_CMP_H
+#define RZ_CMP_H
+
+#include <rz_core.h>
+#include <rz_util.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct {
+	ut32 len; // max 255 bytes diff in one struct
+	bool same;
+	ut8 *data1;
+	ut8 *data2;
+	ut64 addr1;
+	ut64 addr2;
+} RzCompareData;
+
+RZ_API RZ_OWN RzCompareData *rz_cmp_data_data(RZ_NONNULL RzCore *core, ut64 addr1, ut64 addr2, ut32 len);
+RZ_API RZ_OWN RzCompareData *rz_cmp_data_str(RZ_NONNULL RzCore *core, ut64 addr, RZ_NONNULL const ut8 *str, ut32 len);
+RZ_API int rz_cmp_print(RZ_NONNULL RzCore *core, RZ_NONNULL const RzCompareData *cmp, RzComparePrintMode mode);
```

Comment:
```suggestion
RZ_API int rz_cmp_print(RZ_NONNULL RzCore *core, RZ_NONNULL const RzCompareData *cmp, RzCmdStateOutput *state);
```
like all the other `_print` functions. I don't think you need a special enum in this case. Also, by doing so you can directly pass the `state` from the command handler. The parser will take care of allocating the PJ and printing + freeing it as necessary.